### PR TITLE
setup environment variable for debugTarget and launchTarget properly

### PR DESCRIPTION
**Pull request by lygstate**
Signed-off-by: Yonggang Luo <luoyonggang@gmail.com>

### CODE REVIEW

File: src/cmake-tools.ts [hunk 1]
```diff
@@ -1777,6 +1777,15 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
...
     }
+    const debug_environment = [];
+    const debug_env_entries = Object.entries(await drv.getCTestCommandEnvironment());
+    for (const [key, value] of debug_env_entries) {
+      debug_environment.push({
+        name: key,
+        value
+      });
+    }
+    debug_config.environment = debug_environment;
```
Comment (xisui-MSFT) [inline src/cmake-tools.ts:1788]: Nit: feels like Object.entries(...).map() is a more concise way?

Comment (lygstate) [reply src/cmake-tools.ts:1788]: If you insist, otherwise this is more readable:0

Comment (lygstate) [reply src/cmake-tools.ts]: done

**Review by andreeis — APPROVED** [review]

**Comment by bobbrow** [comment]
There is some internal debate as to which preset should supply the environment by default and how we might provide an override for people who don't like that decision.  Right now, Visual Studio uses the configurePreset environment instead of the build or test preset environment.

**Comment by andreeis** [comment]
@lygstate, we may postpone or close this PR since for now we need to implement the same but using the configure preset instead of the test preset. Thank you for creating this PR, it helped us with developing the equivalent that we're going to include in 1.8.
https://github.com/microsoft/vscode-cmake-tools/pull/1977

**Comment by elahehrashedi** [comment]
@lygstate, thank you for your contribution :) The final decision is to follow VS and use configurePresets.

**Pull request closed without merging**